src/components/portrait/portrait.test.js

```diff
@@ -0,0 +1,262 @@
+      }
+    );
+
+    it.each(testImage)(
+      "should check alt text for %s in Portrait component",
+      (src) => {
+        CypressMountWithProviders(<Portrait alt="cypress-test" src={src} />);
+        portraitAlt().should("have.attr", "alt", "cypress-test");
+      }
+    );
+
+    it.each([
+      ["square", "0px"],
+      ["circle", "50%"],
+    ])("should check %s shape for Portrait component", (shape, radius) => {
+      CypressMountWithProviders(<Portrait shape={shape} />);
+      portraitPreview().children().should("have.css", "border-radius", radius);
+    });
+
+    it.each(["error", "warning", "info"])(
+      "should check %s icon for Portrait component",
+      (iconType) => {
+        CypressMountWithProviders(<Portrait iconType={iconType} />);
+        icon().should("have.attr", "data-element", iconType).and("be.visible");
+      }
+    );
+
+    it.each([
+      ["Kate Wool", "KW"],
+      ["Jane Col", "JC"],
+    ])(
+      "should check initials for %s in Portrait component",
+      (name, passInitials) => {
+        CypressMountWithProviders(<Portrait initials={passInitials} />);
+        portraitInitials()
+          .should("have.css", "height", "38px")
+          .and("have.css", "width", "38px");
+      }
+    );
+
+    it.each([
+      [false, "rgb(242, 245, 246)"],
+      [true, "rgb(153, 173, 183)"],
+    ])(
+      "should check when dark background colour set to % in Portrait component",
+      (boolValue, color) => {
+        CypressMountWithProviders(<Portrait darkBackground={boolValue} />);
+        portraitPreview()
+          .children()
+          .should("have.css", "background-color", color);
+      }
+    );
+
+    it.each(testData)(
+      "should check tooltipMessage as %s for Portrait component",
+      (tooltipMessage) => {
+        CypressMountWithProviders(
+          <PortraitComponent tooltipMessage={tooltipMessage} />
+        );
+        portraitPreview().realHover();
+        getDataElementByValue("tooltip")
+          .should("be.visible")
+          .and("have.text", tooltipMessage);
+      }
+    );
+
+    it.each(testData)(
+      "should check tooltipId as %s for Portrait component",
+      (tooltipId) => {
+        CypressMountWithProviders(<PortraitComponent tooltipId={tooltipId} />);
+        portraitPreview().realHover();
+        getDataElementByValue("tooltip")
+          .should("be.visible")
+          .and("have.id", tooltipId);
+      }
+    );
+
+    it.each([true, false])(
+      "should check when tooltip visibility is %s for Portrait component",
+      (boolValue) => {
+        CypressMountWithProviders(
+          <PortraitComponent tooltipIsVisible={boolValue} />
+        );
+        if (boolValue === true) {
+          getDataElementByValue("tooltip")
+            .should("be.visible")
+            .and("have.text", "Rebecca Smith");
+        } else {
+          getDataElementByValue("tooltip").should("not.exist");
+        }
+      }
+    );
```

Comment:
I would recommend splitting this test in two, so one to test when tooltip is visible and one where it is not. That way we would not need the conditional block - which should make the tests easier to debug if they start failing